eth/main.cpp

```diff
@@ -841,6 +848,10 @@ int main(int argc, char** argv)
 //	if (gasFloor != Invalid256)
 //		c_gasFloorTarget = gasFloor;
 
+	if (!chainConfigIsSet)
+		// default to mainnet if not already set with any of `--mainnet`, `--testnet`, `--genesis`, `--config`
+		ChainParams chainParams(genesisInfo(eth::Network::MainNetwork), genesisStateRoot(eth::Network::MainNetwork));
```

Comment:
This won't work, it creates another `chainParams` variable which exists only in the scope of this `if` (so destroyed immediately).

I suggest declaring it as `boost::optional<ChainParams> chainParams;` then you don't need the flag, assign to it as usual `chainParams = ChainParams(...)`, test if it's set with `if (chainParams)` and pass it further as `*chainParams`